problems releasing sourmash v4.2.3

in releasing v4.2.3 (#1761) using https://sourmash.readthedocs.io/en/latest/release.html, I ran into a problem with the first testenv (testenv1) where the `pip install -r requirements.txt` failed. From the error text below, it looks like it is having some trouble finding libpthread.

```
    gcc -pthread -shared -B /home/ctbrown/miniconda3/envs/sourmash-rc/compiler_compat -L/home/ctbrown/miniconda3/envs/sourmash-rc/lib -Wl,-rpath=/home/ctbrown/miniconda3/envs/sourmash-rc/lib -Wl,--no-as-needed -Wl,--sysroot=/ -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,-rpath,/home/ctbrown/miniconda3/envs/sourmash-rc/lib -Wl,-rpath-link,/home/ctbrown/miniconda3/envs/sourmash-rc/lib -L/home/ctbrown/miniconda3/envs/sourmash-rc/lib -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /home/ctbrown/miniconda3/envs/sourmash-rc/include -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /home/ctbrown/miniconda3/envs/sourmash-rc/include build/temp.linux-x86_64-3.8/build/temp.linux-x86_64-3.8/empty.o -o /tmp/tmp.EuG8SHPkQC/testenv1/sourmash/src/sourmash/_lowlevel__lib.so
    /home/ctbrown/miniconda3/envs/sourmash-rc/compiler_compat/ld: cannot find /lib64/libpthread.so.0
    /home/ctbrown/miniconda3/envs/sourmash-rc/compiler_compat/ld: cannot find /usr/lib64/libpthread_nonshared.a
    collect2: error: ld returned 1 exit status
    error: command '/home/ctbrown/miniconda3/envs/sourmash-rc/bin/gcc' failed with exit code 1
```

This looks an awful lot like https://github.com/pandas-dev/pandas/pull/40706#issuecomment-816482976 and https://github.com/ContinuumIO/anaconda-issues/issues/7363, where 'sysroot' is being incorrectly specified as /.

My current theory is that something is wonky with my sourmash-rc conda environment, which was created per our docs like so:
```
conda create -y -n sourmash-rc python=3.8 pip cxx-compiler make twine tox tox-conda setuptools_scm
```
I'm trying again with a python 3.9.environment.

UPDATE: same error as in https://github.com/sourmash-bio/sourmash/issues/1689

--- a/doc/release.md
+++ b/doc/release.md
@@ -53,7 +53,7 @@ git push --tags origin
 
 3\. Test the release candidate. Bonus: repeat on macOS:
 ```
-python -m pip install -U virtualenv wheel
+python -m pip install -U virtualenv wheel tox-setuptools-version
 
 cd ..
 python -m venv testenv1

--- a/doc/release.md
+++ b/doc/release.md
@@ -8,7 +8,7 @@ Michael Crusoe.
 The basic build environment needed below can be created as follows:
 
 ```
-conda create -y -n sourmash-rc python=3.8 pip cxx-compiler make twine tox tox-conda setuptools_scm
+conda create -y -n sourmash-rc python=3.8 pip cxx-compiler make twine tox tox-conda "setuptools<60" setuptools_scm
 ```
 
 Then activate it with `conda activate sourmash-rc`.
@@ -53,7 +53,7 @@ git push --tags origin
 
 3\. Test the release candidate. Bonus: repeat on macOS:
 ```
-python -m pip install -U setuptools pip virtualenv wheel
+python -m pip install -U virtualenv wheel
 
 cd ..
 python -m venv testenv1
@@ -67,7 +67,6 @@ cd testenv1
 source bin/activate
 git clone --depth 1 --branch v${new_version}${rc} https://github.com/sourmash-bio/sourmash.git
 cd sourmash
-python -m pip install -U setuptools pip wheel setuptools_scm
 python -m pip install -r requirements.txt
 make test
 
@@ -76,7 +75,6 @@ make test
 cd ../../testenv2
 deactivate
 source bin/activate
-python -m pip install -U setuptools pip wheel setuptools_scm
 python -m pip install -e git+https://github.com/sourmash-bio/sourmash.git@v${new_version}${rc}#egg=sourmash[test]
 cd src/sourmash
 make test
@@ -89,7 +87,6 @@ cp dist/sourmash*tar.gz ../../../testenv3/
 cd ../../../testenv3/
 deactivate
 source bin/activate
-python -m pip install -U setuptools pip wheel setuptools_scm
 python -m pip install sourmash*tar.gz
 python -m pip install pytest
 tar xzf sourmash-${new_version}${rc}.tar.gz
@@ -113,7 +110,6 @@ Test the PyPI release in a new virtualenv:
 cd ../../testenv4
 deactivate
 source bin/activate
-python -m pip install -U setuptools pip wheel
 # install as much as possible from non-test server!
 python -m pip install screed pytest numpy matplotlib scipy bam2fasta deprecation cffi
 python -m pip install -i https://test.pypi.org/simple --pre sourmash

--- a/doc/release.md
+++ b/doc/release.md
@@ -75,6 +75,7 @@ make test
 cd ../../testenv2
 deactivate
 source bin/activate
+python -m pip install setuptools_scm
 python -m pip install -e git+https://github.com/sourmash-bio/sourmash.git@v${new_version}${rc}#egg=sourmash[test]
 cd src/sourmash
 make test

--- a/tox.ini
+++ b/tox.ini
@@ -16,6 +16,7 @@ isolated_build = true
 skip_missing_interpreters = true
 
 [testenv]
+setuptools_version = setuptools<60
 description = run the tests with pytest under {basepython}
 setenv =
     PIP_DISABLE_VERSION_CHECK = 1

